### DISCUSSION TITLE
Model deviations quite large when performing Molecular Dynamics on training set

### DISCUSSION THREAD

**Discussion by DingChangjie** [deepmodeling/dpgen#419] [Q&A] [answered]
Hi, everyone. I faced some problems when I was trying to decide a proper pair of σ_lo and σ_hi manually (in order to use them in subsequent DpGen) 
I've prepared AIMD data at 0K (and only at 0K), and used them as a whole : I mean by specifying "numb_test": "30%" in the param.json, the program arbitrarily chose 30% frames as testing set. Actually I've done this by the first stage of DpGen, and the four trained models all gave satisfying results. Here is an excerption of an lcurve.out, showing that after 400,000 steps, both energy error and force error had decreased to a low level : 

\# batch      l2_tst    l2_trn    l2_e_tst  l2_e_trn    l2_f_tst  l2_f_trn    l2_v_tst  l2_v_trn         lr
......
 399700    7.77e-03  6.32e-03    7.32e-06  6.31e-06    6.20e-03  4.62e-03    4.85e-04  4.50e-04    1.2e-08
 399800    1.09e-02  1.21e-02    4.81e-06  6.66e-06    7.86e-03  9.07e-03    7.78e-04  8.17e-04    1.2e-08
 399900    8.01e-03  1.03e-02    2.32e-06  2.13e-06    5.84e-03  5.51e-03    5.68e-04  9.02e-04    1.2e-08
 400000    9.73e-03  7.37e-03    6.10e-06  4.08e-06    7.68e-03  5.57e-03    6.21e-04  5.03e-04    1.0e-08

Then I performed Deep Molecular Dynamics simulation on one structure of the training set in order to get a distribution plot of model deviations. Since _fix nvt_ does not allow a temperature range of [0K, 0K], the temperature was set to 5K, considering that a closer temperature to 0K would make more candidates (in official examples σ_lo=0.05 and σ_hi=0.20, which I thought indicates that most trajectories will distribute in this range and be sent to DFT calculations). However I got a distribution plot as below, the data came from model_devi.out directly.
![image](https://user-images.githubusercontent.com/59408640/119223470-407fbb00-bb2c-11eb-9a45-0c37772a8e37.png)
I performed 10,000 MD steps and histogramed every 10 steps. It can be seen that even most of the min force deviations are larger than 0.1 eV/A, not to say the max force deviation. 
Will such deviations be too large ? I'm confused that since the four models all came out of one training set, how could a minor increase of temperature gave such a significant straggling ?(I'm not sure whether I misunderstood something, or there were some problems in preparing the AIMD data...)
Thanks a lot !

**Comment by AnguseZhang** [comment, marked as answer]
1. What is the reason for only performing AIMD at 0K?  The conformations generated this way will be too close to each other, and definitely this will cause a overfitting problem, which may give you an implication that the fitting accuracy of initial data set is high.  
One suggestion is that try to prepare enough representative AIMD trajectories to help you set the trust level for model deviation. Be clear about which condition do you expect your model works in. Sometimes we will increase the trust level as the temperature increases, since the fluctuations will definitely increase. 

2. One other fact is that model deviation is always larger than fitting accuracy (RMSE) according to mathematical formula. 

Best,

Yuzhi

**Comment by DingChangjie** [comment]
PS : I intended to fit a potential suitable for a wide range of temperature (eg 0K to 300K), so I'd like DpGen to explore those configurations at higher temperature, based on AIMD data of 0K

**Comment by DingChangjie** [comment]
@AnguseZhang Thanks for your helpful suggestions ! I'll try to fix this
